src/main/kotlin/network/rs485/logisticspipes/config/ServerConfigurationManager.kt

```diff
@@ -78,11 +78,21 @@ class ServerConfigurationManager {
         return internalRepresentation.playerConfigurations.keys
     }
 
-    fun getPlayerConfiguration(identifier: PlayerIdentifier): ClientConfiguration? {
-        return internalRepresentation.playerConfigurations[identifier] ?: ClientConfiguration()
+    fun getPlayerConfiguration(identifier: PlayerIdentifier): PlayerConfiguration {
+        return internalRepresentation.playerConfigurations[identifier] ?: PlayerConfiguration()
     }
 
     fun setClientConfiguration(identifier: PlayerIdentifier, configuration: ClientConfiguration) {
+        val newConfigurations = HashMap(internalRepresentation.playerConfigurations)
```

Comment:
`internalRepresentation.playerConfigurations.toMutableMap()`?

Reply:
Does toMutableMap copy the original? The name doesn't imply that

Reply:
Yup, it does. I mean, it kinda has to if the original is immutable. In fact, toMap() also copies, as do all the to<Collection> methods

Reply:
alright, then refactor :)